composites/Plugin/Shared/components/Checkbox.js

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+
+const YoastCheckbox = styled.input`
+	margin-right: 0.5em;
+`;
+
+/**
+ * Creates a React Checkbox and Label component.
+ */
+class Checkbox extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			checked: props.checked,
+		};
+	}
+
+	/**
+	 * Handles changes in the Checkbox.
+	 *
+	 * @param {Event} event The onChange event.
+	 * @returns {void} Fires the onChange function passed in the props.
+	 */
+	handleChange( event ) {
+		this.setState( { checked: event.target.checked } );
+		this.props.onChange( this.state.checked );
+	}
+
+	/**
+	 * Renders a checkbox and a label.
+	 *
+	 * @returns {ReactElement} The Checkbox react component including label.
+	 */
+	render() {
+		return(
+			<React.Fragment>
+				<YoastCheckbox type="checkbox" id={ this.props.id } onChange={ this.handleChange.bind( this ) } />
+				<label htmlFor={ this.props.id }>
+					{ this.props.label }
+				</label>
+			</React.Fragment>
+		);
+	}
+}
+
+Checkbox.propTypes = {
+	id: PropTypes.string.isRequired,
+	onChange: PropTypes.func.isRequired,
+	label: PropTypes.oneOfType( [
+		PropTypes.string,
+		PropTypes.array,
+	] ),
+	checked: PropTypes.bool,
+
+};
+
+Checkbox.defaultProps = {
+	checked: false,
+	label: "",
```

Comment:
While _not_passing a label would be very evident because only the checkbox would be visible, it would also print out an empty label element e.g. `<label for="exampleCheckbox"></label>`. We should avoid this and I'd suggest to make the label prop required instead of setting empty string as default value.